custom/enikshay/expressions.py

```diff
@@ -1,10 +1,98 @@
+from jsonobject import DefaultProperty, BooleanProperty
 from jsonobject.properties import ListProperty, StringProperty
 
 from corehq.apps.userreports.expressions.factory import ExpressionFactory
 from corehq.apps.userreports.specs import TypeProperty
+from corehq.form_processor.interfaces.processor import FormProcessorInterface
 from dimagi.ext.jsonobject import JsonObject
 
 
+class FirstCaseFormWithXmlns(JsonObject):
+    type = TypeProperty('first_case_form_with_xmlns')
+    xmlns = StringProperty(required=True)
+    case_id_expression = DefaultProperty(required=True)
+    reverse = BooleanProperty(default=False)
+
+    def configure(self, case_id_expression):
+        self._case_id_expression = case_id_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+
+        if not case_id:
+            return None
+
+        assert context.root_doc['domain']
+        return self._get_forms(case_id, context)
+
+    def _get_forms(self, case_id, context):
+        domain = context.root_doc['domain']
+
+        cache_key = (self.__class__.__name__, case_id, self.xmlns, self.reverse)
+        if context.get_cache_value(cache_key) is not None:
+            return context.get_cache_value(cache_key)
+
+        xforms = FormProcessorInterface(domain).get_case_forms(case_id)
+        xforms = sorted(
+            [form for form in xforms if form.xmlns == self.xmlns and form.domain == domain],
+            key=lambda x: x.received_on
+        )
```

Comment:
Did we consider storing the from XMLNS in the `CaseTransaction` table to allow filtering in the DB?